### DISCUSSION TITLE
Trouble converting EXR to TIF

### DISCUSSION THREAD

**Discussion by TechArtistG** [ImageMagick/ImageMagick#2149] [Help] [answered]
Hi all, I'm using magick mogrify 7.0.10-19 Q16 on Win 10 to convert a series of EXRs to TIF.  
Problem is the resultant TIF files are the correct resolution and bit depth but are just 100% white pixels.  I've checked the source EXRs and they look fine in Photoshop.  One catch is the EXRs are single channel representing a depth map.

Are there any extra options I need to supply magick/mogrify to correctly convert a single channel EXR to floating point TIF?

Thanks!

**Comment by TechArtistG** [comment]
More info: It looks as though the EXR is being clamped at some point in the conversion process even though the final output TIF is 16bit.  I can also confirm that HDRI is enabled in ImageMagick.

**Comment by fmw42** [comment, marked as answer]
Try using magick on one image.

`magick image.exr image.tif`

or

`magick image.exr -define quantum:format=floating-point image.tif`

If that does not help, then post a link to your EXR file.

**Reply by TechArtistG** [reply]
Thanks! Adding -define quantum:format=floating-point and still using mogrify works fine.  However it's weird that ImageMagick detected enough about the images to save the TIF as  16bit but not enough to convert it correctly.

**Comment by fmw42** [comment]
EXR files can have a dynamic range outside the range of 16-bits. The define tells TIFF to save outside the dynamic range if needed and not clip, which it normally would do.

**Reply by TechArtistG** [reply]
Thanks!  Now the only difference I can see between the TIF ImageMagick produces and the one Photoshop produces is when I load the TIFs back into Photoshop, the Photoshop converted one reports "Y, Gray" as channels and the imageMagic one reports "Gray".  Only reason I care is because a the team using the images are loading them into MatLab which is not loading the ImageMagic TIF correctly.

**Reply by fmw42** [reply]
I will have to leave the issue with Matlab to the IM developers.  What is the error you get from Matlab?

**Reply by TechArtistG** [reply]
Ignore the "Y" part, never realized Photoshop puts the selected layer name in the image title.

**Reply by fmw42** [reply]
Can you run

magick identify -verbose image.tif

on both TIFF versions and post the result. Perhaps there are other differences.

**Reply by TechArtistG** [reply]
Comparing  magick identify -verbose outputs of the TIFs generated from from ImaegMagick and Photoshop from the same EXR.  This is using the -define quantum:format=floating-point option.

**IM:**
Image:
  Filename: D:\ServiceData\s-gemof-20200612-022647-msbuilding4\InstanceGroup2DescCameraCamerasx_usd_camera1sx_usd_camera1Shape_iter0\processed\depth\frame_2030049705338974265-sx_usd_camera1Shape_depth_Low_00000.000000.tif
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 5472x3648+0+0
  Units: PixelsPerInch
  Colorspace: Gray
  Type: GrayscaleAlpha
  Base type: Grayscale
  Endianness: LSB
  Depth: 16-bit
  Channel depth:
    Gray: 16-bit
    Alpha: 1-bit
  Channel statistics:
    Pixels: 19961856
    Gray:
      min: 1.21895e+007  (186)
      max: 1.67606e+007 (255.75)
      mean: 1.43112e+007 (218.375)
      standard deviation: 1.22217e+006 (18.6492)
      kurtosis: -1.31772
      skewness: 0.0588496
      entropy: 0
    Alpha:
      min: 65535  (1)
      max: 65535 (1)
      mean: 65535 (1)
      standard deviation: -1.#IND (-1.#IND)
      kurtosis: 2.70805e+057
      skewness: 6.88526e+040
      entropy: 0
  Colors: 559
  Histogram:
    ...
  Rendering intent: Perceptual
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 5472x3648+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Zip
  Orientation: TopLeft
  Convex hull: 0,0 5471,0 5471,3628 5458,3647 0,3647 0,0
  Minimum bounding box: 5471,0 5471,3647 0,3647 0,0
  Properties:
    date:create: 2020-06-15T22:18:16+00:00
    date:modify: 2020-06-15T22:18:18+00:00
    minimum-bounding-box:_p: 0,0
    minimum-bounding-box:_q: 5471,0
    minimum-bounding-box:_v: 5458,3647
    minimum-bounding-box:angle: 0
    minimum-bounding-box:area: 1.99527e+007
    minimum-bounding-box:height: 5471
    minimum-bounding-box:unrotate: -0
    minimum-bounding-box:width: 3647
    quantum:format: floating-point
    signature: 5f7843c8b653ac47627165b2e831c864f6a9c59cdb6f13fa76ef022e0f2c0a6e
    tiff:alpha: unassociated
    tiff:endian: lsb
    tiff:photometric: min-is-black
    tiff:rows-per-strip: 48
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 1.82163MiB
  Number pixels: 19.9619M
  Pixels per second: 28.7023MP
  User time: 8.203u
  Elapsed time: 0:01.695
  Version: ImageMagick 7.0.10-19 Q16 x64 2020-06-12 http://www.imagemagick.org

**Photoshop** (after convert from RGB to Gray):
Image:
  Filename: \\analogfs\vendors\Art\_Projects\3D_reconstruction\SimuatedData\MicrosoftRedmondCampus\EXRtoTIFTest\Camera01_depth_0001_singleChannel_zip.tif
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 5472x3648+0+0
  Resolution: 72x72
  Print size: 76x50.6667
  Units: PixelsPerInch
  Colorspace: Gray
  Type: Grayscale
  Endianness: LSB
  Depth: 32/16-bit
  Channel depth:
    Gray: 16-bit
  Channel statistics:
    Pixels: 19961856
    Gray:
      min: 1.21885e+007  (185.985)
      max: 1.67631e+007 (255.788)
      mean: 1.43112e+007 (218.375)
      standard deviation: 1.22217e+006 (18.6491)
      kurtosis: -1.31772
      skewness: 0.0588524
      entropy: 0
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 5472x3648+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Zip
  Orientation: TopLeft
  Convex hull: 0,0 5471,0 5471,3646 5470,3647 0,3647 0,0
  Minimum bounding box: 5471,0 5471,3647 0,3647 0,0
  Profiles:
    Profile-8bim: 3176 bytes
    Profile-tiff:37724: 22963532 bytes
    Profile-xmp: 47045 bytes
  Properties:
    date:create: 2020-06-11T19:08:15+00:00
    date:modify: 2020-06-11T19:08:09+00:00
    dc:format: image/tiff
    exif:PixelXDimension: 5472
    exif:PixelYDimension: 3648
    minimum-bounding-box:_p: 0,0
    minimum-bounding-box:_q: 5471,0
    minimum-bounding-box:_v: 5470,3647
    minimum-bounding-box:angle: 0
    minimum-bounding-box:area: 1.99527e+007
    minimum-bounding-box:height: 5471
    minimum-bounding-box:unrotate: -0
    minimum-bounding-box:width: 3647
    photoshop:ColorMode: 1
    photoshop:ICCProfile: Linear Grayscale Profile
    quantum:format: floating-point
    signature: f4231845c09504b655ccf27bbf7f248453b4cadf497b777e9717a5f2e101a1ac
    tiff:alpha: unspecified
    tiff:endian: lsb
    tiff:photometric: min-is-black
    tiff:rows-per-strip: 47
    tiff:software: Adobe Photoshop 21.1 (Windows)
    tiff:timestamp: 2020:06:11 12:07:57
    xmp:CreateDate: 2020-06-11T11:18:08-07:00
    xmp:MetadataDate: 2020-06-11T12:07:57-07:00
    xmp:ModifyDate: 2020-06-11T12:07:57-07:00
    xmpMM:DerivedFrom: xmp.did:24b87cf0-7166-6344-89d8-568adb6a150c
    xmpMM:DocumentID: adobe:docid:photoshop:60b39cd0-a907-8049-8f54-666d48a9278f
    xmpMM:InstanceID: xmp.iid:83a6bde6-923e-3144-b112-90e814f9bbb0
    xmpMM:OriginalDocumentID: xmp.did:24b87cf0-7166-6344-89d8-568adb6a150c
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 74.5668MiB
  Number pixels: 19.9619M
  Pixels per second: 8.97725MP
  User time: 1.250u
  Elapsed time: 0:03.223
  Version: ImageMagick 7.0.10-19 Q16 x64 2020-06-12 http://www.imagemagick.org
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1037.
identify: CompressionNotSupported ZipWithPrediction(32 bit) @ error/psd.c/ReadPSDLayer/1501.

**Reply by TechArtistG** [reply]
Differences of note I can see:
IM TIF has a 1bit alpha channel, PS doesn't have an alpha channel
IM Depth = 16bit, PS Depth = 32/16bit
IM tiff:alpha: unassociated, PS tiff:alpha: unspecified
IM tiff:rows-per-strip: 48, PS tiff:rows-per-strip: 47

I think most of the MatLab problem is the Alpha channel

**Reply by fmw42** [reply]
It could also be the 32-bit depth as well. First try removing the alpha channel and see if that works

`magick image.exr -alpha off -define quantum:format=floating-point image.tif`

or

`magick image.exr -alpha disassociate -define quantum:format=floating-point image.tif`

If that does not work, then try

`magick image.exr -alpha off -define quantum:format=floating-point -define quantum:format=signed image.tif`

or

`magick image.exr -alpha disassociate -define quantum:format=floating-point -define quantum:format=signed image.tif`

If that fails, then perhaps you need to install Q32 ImageMagick 7 rather than the default Q16.

**Comment by fmw42** [comment]
I do not know if this matters, but the IM TIFF file is missing the Photoshop profiles. Perhaps Matlab needs one or more of those:

```
Profiles:
Profile-8bim: 3176 bytes
Profile-tiff:37724: 22963532 bytes
Profile-xmp: 47045 bytes

```

**Reply by TechArtistG** [reply]
I know one part is apparently you can't get MatLab to ignore the Alpha.

So I used "-type Grayscale -depth 32 -define quantum:format=floating-point" and that gets me pretty close, removed the alpha channel and Depth = 32/16bit on the IM TIF.  Waiting to hear back if it works in MatLab

**Reply by TechArtistG** [reply]
One interesting thing to note is the origional EXR is about 50MB, the PS TIF(ZIP) is about 70MB and the IM TIF(ZIP) is about 2MB

**Reply by TechArtistG** [reply]
Apparently that worked!  Thanks for your help fmv42!!
